Mathlib/Algebra/Algebra/Quasispectrum.lean

```diff
@@ -370,6 +370,21 @@ lemma quasispectrum_inr_eq (R S : Type*) {A : Type*} [Semifield R]
 
 end Unitization
 
+lemma quasispectrum.swap_mul_eq {R A : Type*} [CommRing R] [NonUnitalRing A] [Module R A]
+    [IsScalarTower R A A] [SMulCommClass R A A] (a b : A) :
+    quasispectrum R (a * b) = quasispectrum R (b * a) := by
+  have (c : A) : ((↑) : Rˣ → R) '' ((↑) ⁻¹' spectrum R (c : Unitization R A)) ∪ {r | ¬ IsUnit r} =
+      spectrum R (c : Unitization R A) := by
+    refine Set.Subset.antisymm (Set.union_subset (by simp) fun r hr ↦ ?_) fun r hr ↦ ?_
+    · simpa [← Unitization.quasispectrum_eq_spectrum_inr] using fun h _ ↦ hr h
+    · by_cases hr' : IsUnit r
+      · lift r to Rˣ using hr'
+        simpa using ⟨r, hr, rfl⟩
+      · exact Or.inr hr'
+  simp only [Unitization.quasispectrum_eq_spectrum_inr]
+  rw [← this (a * b), ← this (b * a), Unitization.inr_mul, Unitization.inr_mul,
+    spectrum.preimage_units_mul_eq_swap_mul]
```

Comment:
I would say we want a version of [spectrum.preimage_units_mul_eq_swap_mul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Spectrum.html#spectrum.preimage_units_mul_eq_swap_mul) staying in `Set R` (with an intersection). So it would be cleaner to first do that, and then the quasispectrum result should just be a few rewrites away thanks to [Set.inter_union_compl](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Basic.html#Set.inter_union_compl) and [quasispectrum.not_isUnit_mem](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Quasispectrum.html#quasispectrum.not_isUnit_mem)